mass-merge.py

```diff
@@ -0,0 +1,203 @@
+#! /usr/bin/env python
+"""
+Merge signatures in bulk, based on an attribute in the spreadsheet.
+
+mass-merge.py takes a list of databases along with spreadsheets (w/-F),
+and merges signatures by values in the specified `merge-col`.
+
+The spreadsheet must contain two columns, 'ident' and the value of `--merge-col`;
+signatures are selected based on 'ident' and renamed to the value found in the merge column.
+Singletons (no additional signatures to merge) will be renamed for consistency.
+"""
+import sys
+import argparse
+import csv
+from collections import defaultdict
+
+import sourmash
+
+from sourmash.picklist import SignaturePicklist
+from sourmash.logging import set_quiet, error, notify, print_results, debug
+from sourmash import sourmash_args
+from sourmash.cli.utils import (add_moltype_args, add_ksize_arg)
+#from sourmash.sig import _check_abundance_compatibility
+
+def _check_abundance_compatibility(sig1, sig2):
+    if sig1.minhash.track_abundance != sig2.minhash.track_abundance:
+        raise ValueError("incompatible signatures: track_abundance is {} in first sig, {} in second".format(sig1.minhash.track_abundance, sig2.minhash.track_abundance))
+
+
+def massmerge(args):
+    """
+    rename one or more signatures.
+    """
+    set_quiet(args.quiet, args.quiet)
+    moltype = sourmash_args.calculate_moltype(args)
+    merge_col = args.merge_col
+
+    # load spreadsheets
+    merge_d = defaultdict(list)
+    all_idents=set()
+    for filename in args.from_spreadsheet:
+        count = 0
+        with open(filename, newline='') as fp:
+            first_entry= True
+            r = csv.DictReader(fp)
+            for row in r:
+                if first_entry:
```

Comment:
probably want to set `first_entry` to False in here, too.

Reply:
resolved via fc16963757a795163d3c17f24c00776012c87d23